### PR TITLE
Using extended Google translate API and comparing synsets of translat…

### PR DESCRIPTION
**Pull request by borealsole**
…ion synonyms with synsets of esv words to find highest Wu-Palmer semantic similarity score (0-1).

Created NMV_translations.json for Genesis 1-2 which contains array of all possible translations returned alongside the farsi word.
Used python NLTK wordnet for matching words as couldn't find a node.js implementation of semantic similarity.
NMV_strongs.json contains esv english words and strongs numbers matched where possible to farsi words for Genesis 1-2.
match is considered when there is a similarity score of 1 between any possible translation synset and any esv word synset.

The matching fails on those criteria for quite a few words that don't find an exact match in the synsets (similarity score of 1) but it is a start.

Sorry it is a bit messy with lots of commented out blocks and I've changed a lot in one commit.

### CODE REVIEW

**Comment by scruffian** [comment]
Thanks for this it looks great! I'll review it properly soon, but the first thing that jumps out at me is that the strong's number should be the second item in the array, not the third.

File: NMV_translations.json [hunk 1]
```diff
@@ -0,0 +1 @@
...,"pillion","refuse","split","craze","cleft","dereliction","chap","desuetude","interstice","proscription","Turkish"]],["کرده،",["relocate","char","do","gig","kick in","perform","ramble","have","joint","rest"]],["به",["to","in","on","into","at","against","quince","quincunx","ac-","ad-","Bah!","Oh!"]],["زن",["woman","female","wife","distaff","carline"]],["خویش",["kindred","relation","connection","connexion","kinswoman","self"]],["خواهد",["want","ask","desire","solicit","wish","require","beg","bone","call","choose","desiderate","intend","invite","invocate","will"]],["پیوست",["annex","appendix","enclosure","annexation","appendage"]],["و",["and"]],["یک",["one","unit","mono-","uni-"]],["تن",["ton","body","person","flesh","corpus"]],["خواهند",["want","ask","desire","solicit","wish","require","beg","bone","call","choose","desiderate","intend","invite","invocate","will"]],["شد.",["become","go","grow","be","happen","leave"]]],[["آدم",[]],["و",["and"]],["زنش",["beat"]],["هر",["any","each","every"]],["دو",["two","twain"]],["عریان",["naked","bare","nude","bald"]],["بودند",["be","stand","exist"]],["و",["and"]],["شرم",["shame","pudency"]],["نداشتند.",["have","possess","own","bear","relieve"]]]]]}}
```
Comment (scruffian) [inline NMV_translations.json:1]: Does this need to be keyed to a book (e.g. Genesis)? I'd have thought it could all just be one top level array.

Comment (scruffian) [reply NMV_translations.json:1]: Sorry ignore this comment

File: wupsimilarity.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+out_json = deepcopy(nmv_translations)
+for book in out_json["books"]:
+    for chapter in range(0, len(out_json["books"][book])):
+        for verse in range(0, len(out_json["books"][book][chapter])):
+            for word in range(0, len(out_json["books"][book][chapter][verse])):
+                if len(out_json["books"][book][chapter][verse][word]) > 1:
+                    out_json["books"][book][chapter][verse][word].pop(1)
+for i, row in similarity_max_df.iterrows():
+    if row["max_similarity"] == 1:
+        data = [row["esv_word"], row["esv_strongs"]]
```
Comment (scruffian) [inline wupsimilarity.py]:
```suggestion
        data = [row["esv_strongs"]]
```

**Comment by scruffian** [comment]
This looks great, are you going to run it for the whole text?

File: wupsimilarity.py (hunk 2 above)
Comment (borealsole) [reply wupsimilarity.py]: Wonder if we can strip out the English later after using it for some sort of QA with a Farsi speaker?

**Comment by borealsole** [comment]
It’s a bit slow at the moment. Would take 3-4 days to do the whole text. Can you see any ways of using promise to speed it up with parallel API calls or anything?

**Comment by scruffian** [comment]
Ah ok, well there's a few options. I think the easiest thing might be to modify the script to take a book as a parameter and then just call it manually 66 times. I'll add that...

File: wupsimilarity.py (hunk 2 above)
Comment (scruffian) [reply wupsimilarity.py]: Well with the strongs numbers embedded, once the text is loaded into sync.bible, it should be easy to see the connection between the farsi and the english, so I think this is unnecessary data.

**Comment by scruffian** [comment]
Ok done that, you can now run it for each book with `node index.mjs Genesis`.

File: NMV_strongs_Genesis.json [hunk 3]
```diff
@@ -0,0 +1 @@
..., "H4325"], ["دیگر", "H5750"], ["هرگز", "H3808"], ["سیل", "H3999"], ["نخواهد"], ["شد", null], ["تا"], ["هر", "H3605"], ["ذی‌جسد"], ["را"], ["هلاک"], ["کند."]], [["هرگاه"], ["رنگین‌کمان"], ["در"], ["ابرها", "H6051"], ["پدیدار"], ["شود،"], ["آن"], ["را"], ["خواهم"], ["دید", "H7200"], ["و"], ["عهد", "H1285"], ["جاودانی"], ["میان", "H996"], ["خدا", "H430"], ["و"], ["هر", "H3605"], ["جاندار", "H2416"], ["ذی‌جسد"], ["را"], ["که"], ["بر", "H5921"], ["زمین", "H776"], ["است"], ["به", "H5921"], ["یاد"], ["خواهم"], ["آورد.»"]], [["پس"], ["خدا", "H430"], ["به", "H5921"], ["نوح"], ["فرمود:", "H559"], ["«این"], ["است"], ["نشان", "H6965"], ["عهدی", "H1285"], ["که"], ["در"], ["میان", "H966"], ["خود"], ["و"], ["هر"], ["ذی‌جسدی"], ["که"], ["بر", "H776"], ["زمین", "H776"], ["است،"], ["استوار"], ["ساخته‌ام.»", "H6965"]], [["پسران", "H1121"], ["نوح"], ["که", null], ["از"], ["کشتی", "H8392"], ["بیرون"], ["آمدند،"], ["سام"], ["و"], ["حام"], ["و"], ["یافِث"], ["بودند.", null], ["حام"], ["پدر", "H1"], ["کنعان", "H3667"], ["بود.", null]], [["اینان"], ["سه", "H7969"], ["پسر", "H1121"], ["نوح"], ["بودند", null], ["و"], ["مردمان"], ["تمامی"], ["جهان", "H776"], ["از"], ["ایشان"], ["منشعب"], ["شدند.", null]]]]}}
```
Comment (scruffian) [inline NMV_strongs_Genesis.json:1]: It would be good to have nothing at all output, rather than `null` as it would save space :)

Comment (borealsole) [reply NMV_strongs_Genesis.json:1]: Oh interesting, hadn’t spotted that! Might be from the pandas dataframe.

**Pull request merged**